sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2073,6 +2073,13 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
               _.containsPattern(FUNCTION_TABLE_RELATION_ARGUMENT_EXPRESSION), ruleId)  {
               case t: FunctionTableSubqueryArgumentExpression =>
                 val alias = SubqueryAlias.generateSubqueryName(s"_${tableArgs.size}")
+                resolvedFunc match {
+                  case Generate(_: PythonUDTF, _, _, _, _, _) =>
+                  case _ if t.hasRepartitioning =>
+                    throw QueryCompilationErrors.tableValuedFunctionPartitionByClauseNotSupported(
```

Comment:
Can we add a test for this case?

Reply:
Python UDTFs are the only type of table function that support input relation arguments right now IIRC in Spark. Certain Spark forks may define their own and then this check would apply. (Please let me know if you can think of any way to test this in this PR.)

Reply:
can we test the builtin TVFs like `explode`?